modules/ROOT/pages/client-credentials.adoc

```diff
@@ -0,0 +1,297 @@
+
+== Request Alias
+
+Some custom OAuth parameters might include characters that are not supported in Java. For example "Api-Key". Since you cannot use "-" as part of a field name, the @OAuthParameter annotation has an optional parameter called requestAlias. For example:
+
+[source, java, linenums]
+----
+@OAuthParameter(requestAlias = "api-key")
+private String apiKey;
+----
+
+=== @OAuthCallbackValue
+
+Callback values are extracted from the response that the service provider sends through the OAuth callback. Although most service providers simply return standard items (such as access and refresh tokens, expiration information, and so on), some others return additional items. In the Salesforce case, they return user and instance IDs.
+
+The annotation includes an expression that is applied on the response to extract the value. That value is then assigned to the field for the connection provider to use. When the connect(), validate() or disconnect() methods are invoked, the fields are set and usable.
+
+=== @ClientCredentialsState
+
+Every ConnectionProvider annotated with @ClientCredentials MUST contain one (and only one) field of type ClientCredentialsState.
+
+It is a simple immutable POJO that contains information regarding the outcome of the OAuth dance. It contains the following information:
+
+[source, java, linenums]
+----
+public interface ClientCrendentialsState {
+
+  /**
+   * @return The obtained access token
+   */
+  String getAccessToken();
+
+  /**
+   * @return The access token's expiration. The actual format of it depends on the OAuth provider
+   */
+  Optional<String> getExpiresIn();
+}
+----
+
+Through this object, the provider gains access to the accessToken and other standard information that was obtained during the OAuth dance. The original Salesforce example shows how the connect() method makes use of this POJO to create the client.
+
+== Configure a Client Credentials Connector
+
+An important aspect of OAuth-enabled modules is how to use them. As much as the SDK does to hide the complexities of the OAuth protocol, things like the OAuth dance are unavoidable. The SDK’s approach to this problem is to standardize the experience of all OAuth modules to keep simplifying the user experience.
+This section discusses the steps a user must take to use the module.
+
+=== Synthetic Parameters
+In addition to everything explicitly defined in the ConnectionProvider, the SDK automatically adds some extra parameters and injects the proper behavior for them.
+
+[cols="1,1,1,2,2", options="header"]
+|===
+|Parameter Name
+|Required
+|Expressions
+|Default Value
+|Description
+
+|`clientId`
+|YES
+|SUPPORTED
+|N/A
```

Comment:
GLOBALLY spell "not available" or "not applicable," or use "n.a." if you must abbreviate, per style ref.
("N/A" literally = "not and/or available/applicable")

Reply:
```suggestion
|N.A.
```